Sources/Presenting/URL/URLManageable.swift

```
@@ -0,0 +1,41 @@
+//
+//  URLManageable.swift
+//
+//  Created by Ibrahim Hamed on 13/01/2024.
+//
+
+import Foundation
+
+/// Protocol defining the requirements for an object to manage URL operations.
+public protocol URLManageable: ObservableObject {
+    /// Configuration for handling URLs. Stores details about how URLs should be opened.
+    var urlConfig: URLConfiguration? { get set }
+
+    /// Opens a URL based on a specified method.
+    ///
+    /// - Parameters:
+    ///   - type: The method for opening the URL (`UrlOpeningType`).
+    ///   - urlString: The URL string to be opened.
+    func open(_ type: URLOpeningType, urlString: String)
```

maybe this should be throwable? 
if the UrlString isn't a URL then it should throw an error